tests/drivers/i2s/i2s_speed/src/test_i2s_codec.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <ztest.h>
+#include <audio/codec.h>
+#include <drivers/i2s.h>
+
+#ifdef CONFIG_MASTER_CLK
+#define CODEC_I2S_OPTIONS (I2S_OPT_FRAME_CLK_SLAVE | I2S_OPT_BIT_CLK_MASTER)
+#else
+#define CODEC_I2S_OPTIONS (I2S_OPT_FRAME_CLK_SLAVE | I2S_OPT_BIT_CLK_SLAVE)
+#endif
+
+#define TIMEOUT     2000
+#define FRAME_CLK_FREQ  8000
+#define CODEC_DEV_NAME DT_LABEL(DT_INST(0, wolfson_wm8960))
+#define SAMPLE_NO   (32)
+#define BLOCK_SIZE (2*SAMPLE_NO)
+
+#ifdef CONFIG_AUDIO_CODEC
+
+int set_codec_loop(void)
+{
+    const struct device *dev_codec = device_get_binding(CODEC_DEV_NAME);
+    struct audio_codec_cfg cfg = {
+        .mclk_freq = 6144000U,
+        .dai_type = AUDIO_DAI_TYPE_I2S,
+        .dai_cfg = {
+            .i2s = {
+                .word_size = 16U,
+                .channels = 2U,
+                .format = I2S_FMT_DATA_FORMAT_I2S,
+                .options = CODEC_I2S_OPTIONS,
+                .frame_clk_freq = FRAME_CLK_FREQ,
+                .block_size = BLOCK_SIZE,
+                .timeout = TIMEOUT,
+                .options = AUDIO_ROUTE_PACKBACK_CAPTURE,
```

Comment:
```suggestion
                .options = AUDIO_ROUTE_PLAYBACK_CAPTURE,
```

Reply:
thanks. update

Reply:
Sorry, but options should really be a set bit flags instead of an enum. Please see how ALSA/ASoC & SOF deal with codec DAI configuration. There is a lot of differentiation in the codec space and also a lot of integration quirks that need to be supported.

Reply:
@lgirdwood  Thanks, there are some merging error, le tme correct it, the options is redefined